flink-clients/src/main/java/org/apache/flink/client/program/rest/retry/ExponentialWaitStrategy.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.client.program.rest.retry;
+
+import static org.apache.flink.util.Preconditions.checkArgument;
+
+/**
+ * {@link WaitStrategy} with exponentially increasing sleep time.
+ */
+public class ExponentialWaitStrategy implements WaitStrategy {
+
+	private final long initialWait;
+
+	private final long maxWait;
+
+	public ExponentialWaitStrategy(final long initialWait, final long maxWait) {
+		checkArgument(initialWait > 0, "initialWait must be positive, was %s", initialWait);
+		checkArgument(maxWait > 0, "maxWait must be positive, was %s", maxWait);
+		this.initialWait = initialWait;
+		this.maxWait = maxWait;
+	}
+
+	@Override
+	public long sleepTime(final long attempt) {
+		final long exponentialSleepTime = initialWait * Math.round(Math.pow(2, attempt));
```

Comment:
`Math.pow(2, attempt)` could also be calculated cheaper with a bit shift operation.

Reply:
`1L` can be only shifted 62 times:

```
	public static void main(String[] args) {
		System.out.println(1L << 62);
		System.out.println(1L << 63);
		System.out.println(1L << 64);
	}
```
output
```
4611686018427387904
-9223372036854775808
1
```